src/H5Fint.c

```diff
@@ -1414,6 +1413,14 @@ H5F__dest(H5F_t *f, bool flush, bool free_on_failure)
             f->shared->efc = NULL;
         } /* end if */
 
+#ifdef H5_HAVE_PARALLEL
+        if (f->shared->mpi_info != MPI_INFO_NULL) {
```

Comment:
I would add a line in `H5F__new` to explicitly set `f->shared->mpi_info` to `MPI_INFO_NULL` when the shared structure is allocated, just in case we fail before the added line that sets this field and we try to call `H5_mpi_info_free` on uninitialized memory. Otherwise, this looks good to me.

Reply:
done